daemon/algod/api/server/v2/handlers.go

```diff
@@ -574,6 +575,10 @@ func (v2 *Handlers) GetProof(ctx echo.Context, round uint64, txid string, params
 		return badRequest(ctx, err, errNoTxnSpecified, v2.Log)
 	}
 
+	if params.Hashtype != nil && *params.Hashtype != "sha512_256" && *params.Hashtype != "sha256" {
```

Comment:
nit: `UnmarshalHashType` and/or `Sha256_512.String()` can be used to simplify some of these checks and leverage the compiler to validate these strings.